pkg/rancher-desktop/components/Snapshots.vue

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+import Vue from 'vue';
+import { mapGetters } from 'vuex';
+
+import EmptyState from '@pkg/components/EmptyState.vue';
+import SnapshotCard from '@pkg/components/SnapshotCard.vue';
+import { Snapshot } from '@pkg/main/snapshots/types';
+
+interface Data {
+  snapshotsPollingInterval: ReturnType<typeof setInterval> | null;
+  isEmpty: boolean;
+}
+
+interface Methods {
+  pollingStart: () => void,
+}
+
+interface Computed {
+  snapshots: Snapshot[]
+}
+
+export default Vue.extend<Data, Methods, Computed, never>({
+  components: { EmptyState, SnapshotCard },
+
+  data(): Data {
+    return {
+      snapshotsPollingInterval: null,
+      isEmpty:                  false,
+    };
+  },
+
+  computed: { ...mapGetters('snapshots', { snapshots: 'list' }) },
+
+  watch: {
+    snapshots: {
+      handler(neu) {
+        this.isEmpty = neu?.length === 0;
+        this.$emit('change', neu);
+      },
+    },
```

Comment:
I think that this watcher would better fit as an `isEmpty` computed prop. Is there a reason why a watcher would be required here?